### ISSUE TITLE
Dependabot not following pagination on quay.io, again

### ISSUE THREAD

**Issue by thomas-riccardi** [dependabot/dependabot-core#10347]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package ecosystem

docker

### Manifest location and content before the Dependabot update

`/Dockerfile`

```dockerfile
FROM quay.io/keycloak/keycloak:24.0.5
...
```

### dependabot.yml content

```yaml
version: 2

updates:
  - package-ecosystem: "docker"
    directory: "/"
    schedule:
      interval: "daily"
```

### Updated dependency

`quay.io/keycloak/keycloak:24.0.5`

### What you expected to see, versus what you actually saw

- expected: `quay.io/keycloak/keycloak:25.0.2` (bumped)
- actual: `quay.io/keycloak/keycloak:24.0.5` (not bumped)

### Images of the diff or a link to the PR, issue, or logs

```
2024/08/01 14:06:47 INFO <job_863701123> Checking all dependencies for version updates...
updater | 2024/08/01 14:06:47 INFO <job_863701123> Checking if keycloak/keycloak 24.0.5 needs updating
  proxy | 2024/08/01 14:06:47 [012] GET https://quay.io:443/v2/keycloak/keycloak/tags/list
  proxy | 2024/08/01 14:06:47 [012] 200 https://quay.io:443/v2/keycloak/keycloak/tags/list
updater | 2024/08/01 14:06:47 INFO <job_863701123> Latest version is 24.0.5
updater | 2024/08/01 14:06:47 INFO <job_863701123> No update needed for keycloak/keycloak 24.0.5
```

**Comment by thomas-riccardi** [comment]
This is the reverse of https://github.com/dependabot/dependabot-core/issues/3989

cf https://github.com/deitch/docker_registry2/issues/106

I have opened a PR on upstream `docker_registry2` lib: https://github.com/deitch/docker_registry2/pull/107

It has been released in 1.18.2: https://github.com/deitch/docker_registry2/releases/tag/v1.18.2

I'm not sure what is the next step to get the fix deployed in dependabot, so I opened this issue.

**Comment by thomas-riccardi** [comment]
`docker_registry2` 1.18.2 is included in https://github.com/dependabot/dependabot-core/pull/10367
